test/e2e/tests/signin-with-ethereum.spec.js

```diff
@@ -0,0 +1,78 @@
+const { strict: assert } = require('assert');
+const { convertToHexValue, withFixtures } = require('../helpers');
+const FixtureBuilder = require('../fixture-builder');
+
+describe('Sign in with ethereum', function () {
+  const ganacheOptions = {
+    accounts: [
+      {
+        secretKey:
+          '0x7C9529A67102755B7E6102D6D950AC5D5863C98713805CEC576B945B15B71EAC',
+        balance: convertToHexValue(25000000000000000000),
+      },
+    ],
+  };
+  it('user should be able to confirm sign in with ethereum', async function () {
+    const expectedSigninMessage =
+      'This site is requesting to sign in with Account 1';
+    const expectedSignInResult =
+      '0xef8674a92d62a1876624547bdccaef6c67014ae821de18fa910fbff56577a65830f68848585b33d1f4b9ea1c3da1c1b11553b6aabe8446717daf7cd1e38a68271c';
+    await withFixtures(
+      {
+        dapp: true,
+        fixtures: new FixtureBuilder()
+          .withPermissionControllerConnectedToTestDapp()
+          .build(),
+        ganacheOptions: {
+          ...ganacheOptions,
+        },
+        title: this.test.title,
+      },
+      async ({ driver }) => {
+        await driver.navigate();
+        await driver.fill('#password', 'correct horse battery staple');
+        await driver.press('#password', driver.Key.ENTER);
+
+        // Create a signin with ethereum request in test dapp
+        await driver.openNewPage('http://127.0.0.1:8080/');
+        await driver.clickElement('#siwe');
+
+        // Wait for signature request popup and check the displayed message
+        await driver.waitUntilXWindowHandles(3);
+        let windowHandles = await driver.getAllWindowHandles();
+        await driver.switchToWindowWithTitle(
+          'MetaMask Notification',
+          windowHandles,
+        );
+
+        const title = await driver.findElement(
+          '.permissions-connect-header__title',
+        );
+        const origin = await driver.findElement('.site-origin');
+        assert.equal(await title.getText(), 'Sign-in request');
+        assert.equal(await origin.getText(), '127.0.0.1:8080');
+
+        const displayedMessage = await driver.findElement(
+          '.permissions-connect-header__subtitle',
+        );
+        const account = await driver.findElement(
+          '.account-list-item__account-name',
+        );
+        assert.equal(
+          `${await displayedMessage.getText()} ${await account.getText()}`,
+          expectedSigninMessage,
+        );
```

Comment:
This looks great, can we add some additional assertions on the message content e.g. Message, URI, Version & Chain etc

Reply:
Added, thanks!